lib/msf/core/web_services/module_search.rb

```diff
@@ -0,0 +1,72 @@
+module Msf::WebServices::ModuleSearch
+
+  def search_modules(opts)
+    raise ::ArgumentError, "At least one search parameter must be provided." if opts.dup.except!(:fields).empty?
```

Comment:
Do we need to call `.dup` on opts here? I don't see the duplicated set getting used or saved anywhere. You should be able to call `.except` (note no `!`) to return the opts hash without `:fields` without making changes to the original.

Reply:
Good catch, thanks!